test/e2e/operator_test.go

```diff
@@ -220,12 +246,35 @@ func TestAWSLoadBalancerControllerWithDefaultIngressClass(t *testing.T) {
 		if err != nil {
 			t.Fatalf("failed to observe the expected log message: %v", err)
 		}
+
+		req, err := http.NewRequest(http.MethodGet, fmt.Sprintf("http://%s", address), nil)
+		if err != nil {
+			t.Fatalf("failed to build client request: %v", err)
+		}
+		req.Host = rule.Host
+
+		var resp *http.Response
+		err = retry.OnError(defaultRetryPolicy,
+			func(err error) bool {
+				return err != nil
+			},
+			func() error {
+				resp, err = httpClient.Do(req)
+				if err == nil {
+					if resp.StatusCode != http.StatusOK {
+						return fmt.Errorf("http status not OK")
+					}
+					return nil
+				} else {
+					return err
+				}
+			})
+		if err != nil {
+			t.Fatalf("failed verify ingress with external client: %v", err)
+		}
+
 		defer func() {
```

Comment:
I just noticed that this defer is called in a for loop which means all of the deferred functions only get called when the surrounding function exits. But this is unrelated to this change.